cobc/codegen.c

```diff
@@ -1072,31 +1078,14 @@ output_base (struct cb_field *f, const cob_u32_t no_output)
 	}
 
 	if (!gen_init_working
+	 && cb_odoslide
 	 && chk_field_variable_address (f)) {
 		if (f01->level == 0
 		 && f01->sister
 		 && strstr (f01->name, " Record")) {	/* Skip to First 01 within FD */
 			f01 = f01->sister;
 		}
-		if (cb_odoslide) {
-			out_odoslide_offset (f01, f);
-		} else {
```

Comment:
Where is that "else" branch gone? Wasn't it executed before?

Reply:
While investigating that code, I had the impression that at that point the `offset` field was already properly set for every field that does not have a varying address, and that the latter case may only occur with `ODOSLIDE` (but still there may be some features I've overlooked).
Furthermore, if I've missed cases that require this specific else branch, then they are not exercised by the test suite: what I suggest is restore the else branch as before and add a comment about this.